src/lib389/lib389/__init__.py

```diff
@@ -938,27 +937,30 @@ def open(self, uri=None, saslmethod=None, sasltoken=None, certdir=None, starttls
             self.log.debug("Using dirsrv ca certificate %s", certdir)
 
         if certdir is not None:
-            """
-            We have a certificate directory, so lets start up TLS negotiations
-            """
+            if not os.access(ensure_str(certdir), os.R_OK):
```

Comment:
I'm not sure this is a good way to check this. It's pretty possible someone has a local instance and is trying to manage a remote one. 

Perhaps a better way is when we parse .dsrc, that sets a flag which is like "dsrc == true" and we can check that to know if the instance is coming from local or remote?

Reply:
certdir is usually tied to the instance directory. So it's highly unlikely that someone has the instances named the same locally and remotely (I'd say it's a bit of an "unhealthy" practice even)

But I agree that it's still a supported option and we shouldn't break that usage...

> Perhaps a better way is when we parse .dsrc, that sets a flag which is like "dsrc == true" and we can check that to know if the instance is coming from local or remote?

I am not sure about that option as we still will encounter this issue:

1. Admin has ~/.dsrc and tries to connect to the remote instance. /etc/openldap/ldap.conf takes care of his certs.
2. We process ~/.dsrc and it doesn't have the cert info (as it's in ldap.conf) and still, we set 'server-id' which then affects the fact that the instance is considered local (hence lib389 looks for certdir instance directory which does not exist on that remote machine)

And it's only one scenario (there are much more things that can go wrong) and it's very cryptic to actually find out if the user is remote or local. So we need a certain and explicit way for specifying is the user wants to connect remotely or locally.

What do you think about the option I mentioned in the issue? 
https://github.com/389ds/389-ds-base/issues/4460#issuecomment-887634353
"When we check for 'isLocal' - check if 'uri' has 'localhost' (and its variations) in it.
If it's localhost or the 'uri' is not defined - we set 'isLocal = True'. And if it's True - then we can get the certdir from the local path (defaults.inf and dse.ldif)"

So I can remove the 'os.access()' part from current PR and I'll add the 'URI' processing so we always have a way to find out if we are local or remote. What do you think?

Reply:
Even checking localhost isn't enough. I have idm.example.com as a localhost alias. And 127.0.0.0/8 maps to your machine. 

I think the only way to undo this mess is that we need to stop using allocate, and have dsctl call "local_simple_allocate" and dsconf/dsidm call "remote_simple_allocate". Then for our CI/tests we have a handler that can do any magic we need.

Reply:
> Even checking localhost isn't enough. I have idm.example.com as a localhost alias. And 127.0.0.0/8 maps to your machine.
> 
> I think the only way to undo this mess is that we need to stop using allocate, and have dsctl call "local_simple_allocate" and dsconf/dsidm call "remote_simple_allocate". Then for our CI/tests we have a handler that can do any magic we need.

I use 'isLocalhost' function, and it tries to resolve the hostname aliases - https://github.com/389ds/389-ds-base/blob/4559a89c0877e45143fa3c9cc34393d4d0cd2696/src/lib389/lib389/utils.py#L675

I was considering using remote_simple_allocate/local_simple_allocate but it really feels like too much risk for 1.4.3 (which is not working correctly now).
I agree that it's better to rewrite our CLI tools but this PR is about fixing the issue for as many possible situations as it can. And I think we should merge it because our users need it.